src/pt/mangaonline/src/eu/kanade/tachiyomi/extension/pt/mangaonline/MangaOnline.kt

```diff
@@ -0,0 +1,229 @@
+
+    override fun latestUpdatesFromElement(element: Element) = SManga.create().apply {
+        title = element.selectFirst("h3 a")!!.ownText()
+            .replace("Capítulo\\s+([\\d.]+)".toRegex(), "")
+            .trim()
+
+        thumbnail_url = element.selectFirst("img")?.absUrl("src")
+
+        val mangaUrl = element.selectFirst("h3 a")!!.absUrl("href")
+            .replace("-capitulo-[\\d-]+".toRegex(), "")
+            .replace("capitulo", "manga")
+
+        setUrlWithoutDomain(mangaUrl)
+    }
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun latestUpdatesRequest(page: Int): Request = throw UnsupportedOperationException()
+
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> {
+        val url = "$baseUrl/capitulo/page/$page".toHttpUrl().newBuilder()
+            .build()
+
+        return client
+            .newCall(GET(url, headers))
+            .asObservableSuccess()
+            .map {
+                val document = it.asJsoup()
+                val mangas = document
+                    .select(latestUpdatesSelector())
+                    .map { latestUpdatesFromElement(it) }
+                    .distinctBy { it.title }
+
+                /*
+                 NOTE: do not navigate to the next page because the manga will be duplicated.
+                 This source launches latest manga chapters unbundled.
+                 */
+                MangasPage(mangas, false)
+            }
+    }
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        val containerInfo = document.selectFirst("div.content > div.sheader")
+        title = containerInfo!!.selectFirst("h1")!!.ownText()
+        thumbnail_url = containerInfo.selectFirst("img")?.absUrl("src")
+        description = containerInfo.selectFirst("p:last-child")?.ownText()
+        genre = containerInfo.select("div.sgeneros a")
+            .map { it.ownText() }
+            .filter { it.length > MIN_LENGTH_GENRER }
+            .joinToString()
+        setUrlWithoutDomain(document.location())
+    }
+
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select("img[loading=lazy]").mapIndexed { i, it ->
+            Page(i, imageUrl = it.absUrl("src"))
+        }
+    }
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        title = element.selectFirst("h3 a")!!.ownText()
+        thumbnail_url = element.selectFirst("img")?.absUrl("src")
+        setUrlWithoutDomain(element.selectFirst("h3 a")!!.absUrl("href"))
+    }
+
+    override fun popularMangaNextPageSelector() = null
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = "$baseUrl/mais-vistos/".toHttpUrl()
+            .newBuilder()
+            .build()
+        return GET(url, headers)
```

Comment:
```suggestion
        val url = "$baseUrl/mais-vistos/"
        return GET(url, headers)
```
httpurl builder not needed here

Reply:
Resolved: [emove URLBuilder](https://github.com/keiyoushi/extensions-source/pull/1566/commits/90e82953f9ba8529a57959f2dca0925d0e33c6d6)